DEV.md

```diff
@@ -0,0 +1,232 @@
+    ** Actions are stateless; state is factored into a separate "TransformerState" object which is passed as a parameter to actions.
+    ** An action may (but usually does not) change the name of a resource which is being transformed.
+    ** An action may change the contents of a target resource.
+
+  * Actions record and report on their effects:
+    ** Each action provides a change object which records, minimally, the count of changes, and whether the target resource name was changed.
+    ** An action may specialize the type of change object that it uses.
+
+  * Actions are driven by configuration data, which is supplied by properties files.
+    ** Property values usually encode an 'initial value - final value' pair, with the initial value as the property name and with the final value as the property value.
+    ** Other property values encode more detailed rule data.
+    ** Property data is not used in a one-to-one correspondence with actions.  What property data is used by which actions is peculiar to the several types of actions.
+
+  * Action types are broadly partitioned into "container type actions", "non-container (leaf) type actions, and a special "null" action.
+
+  * Container actions are responsible for iterating across their contents, and recursively invoking the action API on each container element.
+    ** Container actions each has an ordered collection of nested actions, which are used to process the elements of the container.
+    ** Container actions provide input stream to a recursively invoked action.
+    ** For best performance, container actions are strongly encouraged to use streaming to process container elements.
+    ** Streaming avoids the creation of temporary files.
+
+   * Leaf actions are responsible for reading, transforming, and writing the contents of a specific file of the type handled by the type of action.
+     ** Leaf actions are invoked with either a file or an input stream.
+     *** An action may decline to process a specific file.
+
+   * The null action handles file types which are otherwise not handled by any type of action, and performs a simple pass-through of the input data.
+
+++ Java and Command Line APIs
+
+Core APIs consist of the following:
+
+* Class org.eclipse.transformer.Transformer" provides a standard "main" command line entry point.
+
+* Class org.eclipse.transformer.Transformer" also provides for direct invocation from java using
+  "int Transformer.runWith(PrintStream, PrintStream, String[])".
+
+A third API is provided for Jakarta type transformations:
+
+* Class org.eclipse.transformer.jakarta.JakartaTransformer" provides an addition "main" command line entry point.  "JakartaTransformer" is packaged with default update properties files and supplies these as default property files for the several update properties files.
+
+package org.eclipse.transformer;
```

Comment:
Use a fenced code block around the code https://github.github.com/gfm/#fenced-code-blocks

~~~markdown
```java
javacode
```
~~~

Reply:
Corrected by refresh of the files.